docs/latest/tutorial/tutorial-main-renderer.md

```diff
@@ -0,0 +1,189 @@
+
+We are going to creae a `preload` script that exposes the versions of Chrome, Node, and
+Electron into the renderer. To do this, add a new file `preload.js` with the
+following:
+
+```js title="preload.js"
+const { contextBridge } = require('electron')
+
+contextBridge.exposeInMainWorld('versions', {
+  node: () => process.versions.node,
+  chrome: () => process.versions.chrome,
+  electron: () => process.versions.electron
+})
+```
+
+The above code accesses the Node.js `process.versions` object and exposes some of them in
+the renderer process in a `global` object called `versions`.
+
+To attach this script to your renderer process, pass in the path to your preload script
+to the `webPreferences.preload` option in the `BrowserWindow` constructor:
+
+```js {9} title="main.js"
+const { app, BrowserWindow } = require('electron')
+const path = require('path')
+
+const createWindow = () => {
+  const win = new BrowserWindow({
+    width: 800,
+    height: 600,
+    webPreferences: {
+      preload: path.join(__dirname, 'preload.js'),
+      sandbox: true
+    }
+  })
+
+  win.loadFile('index.html');
+}
+
+app.whenReady().then(() => {
+  createWindow()
+})
+```
+
+:::tip
+There are two Node.js concepts that are used here:
+
+- The [`__dirname`][dirname] string points to the path of the currently executing script
+  (in this case, your project's root folder).
+- The [`path.join`][path-join] API joins multiple path segments together, creating a
+  combined path string that works across all platforms.
+
+If you are not familiar with Node.js, we recommend you to read [this guide][node-guide].
+:::
+
+At this point, the renderer context has access to `versions`, so let's make that information
+available to the user. Create a new file `renderer.js` that contains the following code:
+
+```js title="renderer.js"
+const information = document.getElementById('information')
+information.innerText = `This app is using Chrome (v${versions.chrome()}), Node.js (v${versions.node()}), and Electron (v${versions.electron()})`
+```
+
+And modify `index.html` to match the following:
+
+```html {18,20} title="index.html"
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="UTF-8" />
+    <meta
+      http-equiv="Content-Security-Policy"
+      content="default-src 'self'; script-src 'self'"
+    />
+    <meta
+      http-equiv="X-Content-Security-Policy"
+      content="default-src 'self'; script-src 'self'"
+    />
+    <title>Hello from Electron renderer!</title>
+  </head>
+  <body>
+    <h1>Hello from Electron renderer!</h1>
+    <p>👋</p>
+    <p id="info"></p>
+  </body>
+  <script src="./renderer.js"></script>
+</html>
+```
+
+We have created a paragraph with `info` as its id, and included `renderer.js`.
+
+:::warning
+The reason why we do not write the JavaScript directly in the HTML between `<script></script>`
+is because we have a Content-Security-Policy that will prevent its execution.
+To know more you can visit [MDN's CSP content][mdn-csp].
+:::
```

Comment:
```suggestion
The reason why we do not write the JavaScript directly in the HTML between `<script></script>`
is because we have a Content Security Policy (CSP) that will prevent its execution.
To know more you can visit [MDN's CSP documentation][mdn-csp].
:::
```